eng/tools/suppressions/test/e2e.test.ts

```diff
@@ -0,0 +1,38 @@
+import { expect, test } from "vitest";
+import { getSuppressions } from "../src/suppressions.js";
+import { join } from "path";
+
+async function getTestSuppressions(path: string, tool: string = "TestTool") {
+  return await getSuppressions(tool, join(__dirname, "e2e", path));
+}
+
+test.concurrent("no suppressions.yaml", async () => {
+  const suppressions = await getTestSuppressions("noSuppressionsYaml");
+  expect(suppressions).toEqual([]);
+});
+
+test.concurrent("empty suppressions.yaml", async () => {
+  const suppressions = await getTestSuppressions("emptySuppressionsYaml");
+  expect(suppressions).toEqual([]);
+});
+
+test.concurrent("suppress foo.json, get foo.json", async () => {
+  const suppressions = await getTestSuppressions(join("suppressFooJson", "foo.json"));
+  expect(suppressions).toEqual([
+    {
+      tool: "TestTool",
+      paths: ["foo.json"],
+      reason: "test",
+    },
+  ]);
+});
+
+test.concurrent("suppress foo.json, get bar.json", async () => {
+  const suppressions = await getTestSuppressions(join("suppressFooJson", "bar.json"));
+  expect(suppressions).toEqual([]);
+});
+
+test.concurrent("suppress foo.json, get foo.json w/ different tool", async () => {
+  const suppressions = await getTestSuppressions(join("suppressFooJson", "foo.json"), "TestTool2");
+  expect(suppressions).toEqual([]);
+});
```

Comment:
Supposedly one of the recommended practices is to wrap all the tests into [`describe` clause](https://vitest.dev/api/#describe), which I guess mimics "test suite" or "test set". I am OK both ways, but I think it would be cool if we would experiment with both approach and see what are the pros/cons.

Reply:
I didn't see a strong use for the describe clause given the number of tests, so I left it out, figuring simpler is better.  We could try requiring a describe clause, if we can automate enforcement.  I really want to avoid needing to **remember** a bunch of finicky rules for our internal eng code.